builtin/providers/aws/resource_aws_iam_role.go

```diff
@@ -23,14 +24,17 @@ func resourceAwsIamRole() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
 			"unique_id": &schema.Schema{
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
 			"name": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:          schema.TypeString,
```

Comment:
Again this needs to be computed so that when we don't set it as part of our config, the Read can still set it and allow it to not cause Updates :)